File: 499847c.py
def dist(a, b):
    return (b[0] - a[0]) ** 2 + (b[1] - a[1]) ** 2

def midp(a, b):
    return ((b[0] + a[0]) / 2.0, (b[1] + a[1]) / 2.0)

def are_close(a, b, tol=1e-9):
    return abs(a - b) < tol

if __name__ == "__main__":
    n = int(input())
    
    v = []
    for _ in range(n):
        a, b = map(int, input().split())
        v.append((a, b))
    
    count = 0
    for i in range(n):
        for j in range(n):
            if j != i:
                for k in range(j + 1, n):
                    if k != i:
                        midpoint = midp(v[j], v[k])
                        if are_close(midpoint[0], v[i][0]) and are_close(midpoint[1], v[i][1]):
                            count += 1

    print(count )  

File: 499847c.cpp
#include<bits/stdc++.h>
using namespace std;
long dist(pair<long,long> a, pair<long,long> b){
    return ((b.first-a.first)*(b.first-a.first))+((b.second-a.second)*(b.second-a.second));
}


pair<long,long> midp(pair<long,long> a, pair<long,long> b){
    return make_pair((b.first + a.first) , (b.second + a.second));
}

int main(){
    int n;
    cin>>n;
    vector<pair<long,long>> v;
    for(int i=0;i<n;i++){
        long a,b;
        cin>>a>>b;
        v.push_back(make_pair(a,b));
    }
    
    int count=0;
    for(int i=0;i<n;i++){
        for(int j=0;(j<n && j!=i);j++){
            for(int k=j+1;(k<n && k!=i);k++){
                if(midp(v[j],v[k]).first==2*v[i].first and midp(v[j],v[k]).second==2*v[i].second){       // if i is midpoint of j and k 
                    count++;
                }  
            }
        }
    }
    cout<<count<<endl;
}